x-pack/plugins/maps/public/components/toolbar_overlay/view.js

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonIcon,
+  EuiPopover,
+  EuiContextMenu,
+  EuiSelectable,
+  EuiHighlight,
+  EuiTextColor,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { getIndexPatternsFromIds } from '../../index_pattern_util';
+import _ from 'lodash';
+import { DRAW_TYPE } from '../../actions/store_actions';
+import { ES_GEO_FIELD_TYPE } from '../../../common/constants';
+
+const RESET_STATE = {
+  isPopoverOpen: false,
+  drawType: null
+};
+
+export class ToolbarOverlay extends React.Component {
+
+
+  state = {
+    isPopoverOpen: false,
+    uniqueIndexPatternsAndGeoFields: [],
+    drawType: null
+  };
+
+  _openToolbar = () => {
```

Comment:
Can you rename. This function is used for opening and closing. Maybe rename to `_togglePopover`